src/scheduler/fifo.c

```diff
@@ -2487,14 +2484,11 @@ parse_sched_obj(struct batch_status *status)
 	char *save_ptr;
 	int i;
 	int j;
-	static int firsttime = 0;
+	long prev_attr_u_period = sc_attrs.attr_update_period;
 
 	attrp = status->attribs;
 
-	if (firsttime) {
-		init_sc_attrs();
-		firsttime = 0;
-	}
+	init_sc_attrs();
```

Comment:
Not sure I understand, the attrs are only updated on qmgr command runs right (not in main sched loop)? We need to do this every time because we don't set "default" value for all sched attributes when we unset them. So, when something is unset, we might not update the respective attribute unless we call this function.

Reply:
I'm not talking about main_sched_loop().  I'm talking about the loop in main() in pbs_sched.c.  The first thing it does is make a call to update the scheduler attrs.  This is meant to happen only on the first cycle after we start.  This is what firstcycle is meant for.  It's so that call just does something on the first cycle.

Reply:
ah, ok, thanks for clarifying. But I think we still need to call init_sc_attrs() every time a qmgr operation on sched object is done. init_sc_attrs() just resets all sc_attrs to their default, so that if any sched attribute has been unset, sched will set it to its default. Take job_sort_formula_threshold for example, it doesn't have a default. So, if admin unsets it, the batch_status list will not have a default value of job_sort_formula_threshold, so if we don't call init_sc_attrs() then we will leave it at its old value. Does that make sense?